src/praisonai/praisonai/sandbox/modal.py

```
@@ -0,0 +1,403 @@
+                )
+                gpu_config = None
+            
+            # Create function for code execution
+            @self._app.function(
+                image=image,
+                gpu=gpu_config,
+                timeout=self.timeout,
+                allow_concurrent_inputs=10,
+                keep_warm=1,
+            )
+            def execute_code(code: str, language: str = "python", env_vars: Dict[str, str] = None):
```

Ruff (0.15.10)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)